src/engine/Input/PointerEventReceiver.ts

```diff
@@ -0,0 +1,420 @@
+  // TODO make this private
+  public currentFrameDown: PointerEvent[] = [];
+  public currentFrameUp: PointerEvent[] = [];
+  public currentFrameMove: PointerEvent[] = [];
+  public currentFrameCancel: PointerEvent[] = [];
+  public currentFrameWheel: WheelEvent[] = [];
+
+  constructor(public readonly target: GlobalEventHandlers & EventTarget, public engine: Engine) {
+    super();
+  }
+
+  private _pointers: PointerAbstraction[] = [this.primary];
+  public at(index: number): PointerAbstraction {
+    if (index >= this._pointers.length) {
+      // Ensure there is a pointer to retrieve
+      for (let i = this._pointers.length - 1, max = index; i < max; i++) {
+        this._pointers.push(new PointerAbstraction());
+      }
+    }
+    return this._pointers[index];
+  }
+
+  public count(): number {
+    return this._pointers.length;
+  }
+
+  public isDown(pointerId: number) {
+    return this.currentFramePointerDown.get(pointerId) ?? false;
+  }
+
+  public wasDown(pointerId: number) {
+    return this.lastFramePointerDown.get(pointerId) ?? false;
+  }
+
+  /**
+   * Whether the Pointer is currently dragging.
+   */
+  public isDragging(pointerId: number): boolean {
+    return this.isDown(pointerId);
+  }
+
+  /**
+   * Whether the Pointer just started dragging.
+   */
+  public isDragStart(pointerId: number): boolean {
+    return this.isDown(pointerId) && !this.wasDown(pointerId);
+  }
+
+  /**
+   * Whether the Pointer just ended dragging.
+   */
+  public isDragEnd(pointerId: number): boolean {
+    return !this.isDown(pointerId) && this.wasDown(pointerId);
+  }
+
+  on(event: 'move', handler: (event: PointerEvent) => void): void;
+  on(event: 'down', handler: (event: PointerEvent) => void): void;
+  on(event: 'up', handler: (event: PointerEvent) => void): void;
+  on(event: 'wheel', handler: (event: WheelEvent) => void): void;
+  on(event: string, handler: (event: any) => void): void {
+    super.on(event, handler);
+  }
+
+  once(event: 'move', handler: (event: PointerEvent) => void): void;
+  once(event: 'down', handler: (event: PointerEvent) => void): void;
+  once(event: 'up', handler: (event: PointerEvent) => void): void;
+  once(event: 'wheel', handler: (event: WheelEvent) => void): void;
+  once(event: string, handler: (event: any) => void): void {
+    super.once(event, handler);
+  }
+
+  off(event: 'move', handler?: (event: PointerEvent) => void): void;
+  off(event: 'down', handler?: (event: PointerEvent) => void): void;
+  off(event: 'up', handler?: (event: PointerEvent) => void): void;
+  off(event: 'wheel', handler?: (event: WheelEvent) => void): void;
+  off(event: string, handler?: (event: any) => void): void {
+    super.off(event, handler);
+  }
+
+  public update() {
+    this.lastFramePointerDown = new Map(this.currentFramePointerDown);
+    this.lastFramePointerPosition = new Map(this.currentFramePointerPositions);
+
+    for (const event of this.currentFrameDown) {
+      this.emit('down', event as any);
```

Comment:
Anything more narrow than `any` for all of these?

Reply:
The bummer with the current EventDispatcher is it requires a GameEvent, I'll try tweaking it to remove the `any`'s

Reply:
Doh, the any's here were unnecessary good find